CONTRIBUTING.md

```diff
@@ -117,22 +69,70 @@ repo root, run:
 ```sh
 mkdir build
 cd build
-cmake -Duse_default_uuid=ON ..
+cmake ..
 cmake --build .
 ```
+
+#### CMake build options
+The following CMake options are available for adding/removing project features.
+
+<table>
+<tr>
+<td>Option</td>
+<td>Description</td>
+<td>Default Value</td>
+</tr>
+<tr>
+<td>BUILD_TESTING</td>
+<td>Generates Unit Test for compilation. CMake will automatically download and build g-test.<br>After Compiling, use `ctest` to run Unit Test.</td>
+<td>OFF</td>
+</tr>
+<tr>
+<td>BUILD_STORAGE_SAMPLES</td>
+<td>Build Azure Storage clients sample application.</td>
+<td>OFF</td>
+</tr>
+<tr>
+<td>RUN_LONG_UNIT_TESTS</td>
+<td>Enables the special unit tests which takes more than 3 minutes to run. THis tests are for some specific features like the connection pool for curl transport adapter.</td>
+<td>OFF</td>
+</tr>
+<tr>
+<td>WARNINGS_AS_ERRORS</td>
+<td>Warnings will make compiling fail</td>
+<td>ON</td>
+</tr>
+<tr>
+<td>BUILD_CURL_TRANSPORT</td>
+<td>Build the curl http transport adapter. When building on Posix systems, if no other transport adapter is built, this option will be automatically turned ON</td>
+<td>OFF</td>
+</tr>
+<tr>
+<td>BUILD_DOCUMENTATION</td>
```

Comment:
Why do we have this option? Can't we do the same thing as C and run `doxygen Doxyfile`?

Reply:
Not sure who added this.
We can review it apart from this PR. Then, if we don't want it, we need a PR to remove it.
Maybe Daniel added for CI release or something. I am not sure.  At any case, it was not documented in the ReadMe